### PR TITLE
python3Packages.surepy: 0.5.0 -> 0.6.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.6.0

Change log: https://github.com/benleb/surepy/releases/tag/v0.6.0

Requires later `rich` release.
- https://github.com/willmcgugan/rich/releases/tag/v10.1.0
- https://github.com/willmcgugan/rich/releases/tag/v10.0.1
- https://github.com/willmcgugan/rich/releases/tag/v10.0.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
`labelbox` failure will be fixed with #121677.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 121732` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.shap</li>
    <li>python38Packages.skorch</li>
    <li>python38Packages.textacy</li>
    <li>python39Packages.optuna</li>
    <li>python39Packages.skorch</li>
    <li>python39Packages.textacy</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>backblaze-b2</li>
    <li>conan</li>
    <li>nrfutil</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.labelbox</li>
    <li>python38Packages.pywick</li>
    <li>python38Packages.sunpy</li>
    <li>python39Packages.labelbox</li>
    <li>python39Packages.pywick</li>
    <li>python39Packages.sunpy</li>
  </ul>
</details>
<details>
  <summary>178 packages built:</summary>
  <ul>
    <li>aiodnsbrute</li>
    <li>ansible-lint (python38Packages.ansible-lint)</li>
    <li>check_systemd</li>
    <li>checkov</li>
    <li>deja-dup</li>
    <li>dpt-rp1-py</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>evdevremapkeys</li>
    <li>evillimiter</li>
    <li>ffmpeg-normalize</li>
    <li>gdown (python38Packages.gdown)</li>
    <li>gnome-keysign</li>
    <li>home-assistant</li>
    <li>jellyfin-mpv-shim</li>
    <li>kaggle (python38Packages.kaggle)</li>
    <li>keepkey_agent (python38Packages.keepkey_agent)</li>
    <li>ldeep</li>
    <li>ledger_agent (python38Packages.ledger_agent)</li>
    <li>lieer</li>
    <li>luigi</li>
    <li>magic-wormhole (python38Packages.magic-wormhole)</li>
    <li>manim</li>
    <li>mathlibtools (python38Packages.mathlibtools)</li>
    <li>mnemosyne</li>
    <li>noto-fonts-emoji</li>
    <li>ocrmypdf</li>
    <li>papis (python38Packages.papis)</li>
    <li>past-time</li>
    <li>piston-cli</li>
    <li>plex-mpv-shim</li>
    <li>python38Packages.afdko</li>
    <li>python38Packages.aiosmb</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.ansible-runner</li>
    <li>python38Packages.b2sdk</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.cirq</li>
    <li>python38Packages.cve-bin-tool</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.dipy</li>
    <li>python38Packages.dnachisel</li>
    <li>python38Packages.enrich</li>
    <li>python38Packages.ffmpeg-progress-yield</li>
    <li>python38Packages.genome-collector</li>
    <li>python38Packages.gradient</li>
    <li>python38Packages.gradient-utils</li>
    <li>python38Packages.graphtage</li>
    <li>python38Packages.gtts</li>
    <li>python38Packages.habanero</li>
    <li>python38Packages.holoviews</li>
    <li>python38Packages.huggingface-hub</li>
    <li>python38Packages.hvplot</li>
    <li>python38Packages.hyperopt</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.internetarchive</li>
    <li>python38Packages.libagent</li>
    <li>python38Packages.lsassy</li>
    <li>python38Packages.mesa</li>
    <li>python38Packages.moviepy</li>
    <li>python38Packages.msldap</li>
    <li>python38Packages.myfitnesspal</li>
    <li>python38Packages.nagiosplugin</li>
    <li>python38Packages.nbsmoke</li>
    <li>python38Packages.ndjson</li>
    <li>python38Packages.nltk</li>
    <li>python38Packages.nototools</li>
    <li>python38Packages.panel</li>
    <li>python38Packages.papermill</li>
    <li>python38Packages.parfive</li>
    <li>python38Packages.perfplot</li>
    <li>python38Packages.plexapi</li>
    <li>python38Packages.proglog</li>
    <li>python38Packages.pypiserver</li>
    <li>python38Packages.pyppeteer</li>
    <li>python38Packages.pypykatz</li>
    <li>python38Packages.pyro-ppl</li>
    <li>python38Packages.python-daemon</li>
    <li>python38Packages.python-miio</li>
    <li>python38Packages.python-mpv-jsonipc</li>
    <li>python38Packages.pytorch-lightning</li>
    <li>python38Packages.rbtools</li>
    <li>python38Packages.rich</li>
    <li>python38Packages.rising</li>
    <li>python38Packages.sacremoses</li>
    <li>python38Packages.salmon-mail</li>
    <li>python38Packages.scancode-toolkit</li>
    <li>python38Packages.surepy</li>
    <li>python38Packages.thinc</li>
    <li>python38Packages.toggl-cli</li>
    <li>python38Packages.tqdm</li>
    <li>python38Packages.transformers</li>
    <li>trezor_agent (python38Packages.trezor_agent)</li>
    <li>python38Packages.tubeup</li>
    <li>python38Packages.twine</li>
    <li>python39Packages.afdko</li>
    <li>python39Packages.aiosmb</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.ansible-lint</li>
    <li>python39Packages.ansible-runner</li>
    <li>python39Packages.b2sdk</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.cirq</li>
    <li>python39Packages.cve-bin-tool</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.dipy</li>
    <li>python39Packages.dnachisel</li>
    <li>python39Packages.enrich</li>
    <li>python39Packages.ffmpeg-progress-yield</li>
    <li>python39Packages.gdown</li>
    <li>python39Packages.genome-collector</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.gradient-utils</li>
    <li>python39Packages.graphtage</li>
    <li>python39Packages.gtts</li>
    <li>python39Packages.habanero</li>
    <li>python39Packages.holoviews</li>
    <li>python39Packages.huggingface-hub</li>
    <li>python39Packages.hvplot</li>
    <li>python39Packages.hyperopt</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.internetarchive</li>
    <li>python39Packages.kaggle</li>
    <li>python39Packages.keepkey_agent</li>
    <li>python39Packages.ledger_agent</li>
    <li>python39Packages.libagent</li>
    <li>python39Packages.lsassy</li>
    <li>python39Packages.magic-wormhole</li>
    <li>python39Packages.mathlibtools</li>
    <li>python39Packages.mesa</li>
    <li>python39Packages.moviepy</li>
    <li>python39Packages.msldap</li>
    <li>python39Packages.myfitnesspal</li>
    <li>python39Packages.nagiosplugin</li>
    <li>python39Packages.nbsmoke</li>
    <li>python39Packages.ndjson</li>
    <li>python39Packages.nltk</li>
    <li>python39Packages.nototools</li>
    <li>python39Packages.panel</li>
    <li>python39Packages.papermill</li>
    <li>python39Packages.papis</li>
    <li>python39Packages.parfive</li>
    <li>python39Packages.perfplot</li>
    <li>python39Packages.plexapi</li>
    <li>python39Packages.proglog</li>
    <li>python39Packages.pypiserver</li>
    <li>python39Packages.pyppeteer</li>
    <li>python39Packages.pypykatz</li>
    <li>python39Packages.pyro-ppl</li>
    <li>python39Packages.python-daemon</li>
    <li>python39Packages.python-miio</li>
    <li>python39Packages.python-mpv-jsonipc</li>
    <li>python39Packages.pytorch-lightning</li>
    <li>python39Packages.rbtools</li>
    <li>python39Packages.rich</li>
    <li>python39Packages.rising</li>
    <li>python39Packages.sacremoses</li>
    <li>python39Packages.salmon-mail</li>
    <li>python39Packages.scancode-toolkit</li>
    <li>python39Packages.surepy</li>
    <li>python39Packages.thinc</li>
    <li>python39Packages.tqdm</li>
    <li>python39Packages.transformers</li>
    <li>python39Packages.trezor_agent</li>
    <li>python39Packages.tubeup</li>
    <li>python39Packages.twine</li>
    <li>r128gain</li>
    <li>sacad</li>
    <li>spotdl</li>
    <li>tartube</li>
    <li>theharvester</li>
    <li>tinyprog</li>
    <li>tockloader</li>
    <li>tts</li>
    <li>twitter-color-emoji</li>
    <li>urlwatch</li>
  </ul>
</details>

File: pkgs/development/python-modules/rich/default.nix [hunk 1]
```diff
@@ -1,40 +1,46 @@
...
+, fetchFromGitHub
 , ipywidgets
-, poetry
+, poetry-core
 , pygments
-, typing-extensions
 , pytestCheckHook
+, pythonOlder
+, typing-extensions
 }:
```
Comment (mweinelt) [inline pkgs/development/python-modules/rich/default.nix:13]: Some people sort these packages by the order they are used in, so I wouldn't apply sorting here. Also makes diffs harder to read.

**Comment by risicle** [comment]
:+1: No new build failures for me, nixos x86_64, but would be nice to get a darwin tester as ever.

**Comment by mweinelt** [comment]
I'll need this for the home-assistant bump later today, so I can't wait for a darwin reviewer to appear out of thin air I'm afraid.

**Comment by mweinelt** [comment]
The `postPatch` phase in surepy looks outdated.

https://github.com/benleb/surepy/commit/f0bfffa32aaf07e5285db68395d64f59e6ca9873

**Comment by fabaff** [comment]
> The postPatch phase in surepy looks outdated.

Thanks, `postPatch` removed.

**Comment by mweinelt** [comment]
Please enable home-assistant tests and rebase on top of master.

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 121732` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>32 packages marked as broken and skipped:</summary>
  <ul>
    <li>evillimiter</li>
    <li>jellyfin-mpv-shim</li>
    <li>python38Packages.baselines</li>
    <li>python38Packages.ignite</li>
    <li>python38Packages.labelbox</li>
    <li>python38Packages.ndjson</li>
    <li>python38Packages.optuna</li>
    <li>python38Packages.pymc3</li>
    <li>python38Packages.pyro-ppl</li>
    <li>python38Packages.pytorch-lightning</li>
    <li>python38Packages.pytorch-metric-learning</li>
    <li>python38Packages.pywick</li>
    <li>python38Packages.rising</li>
    <li>python38Packages.scancode-toolkit</li>
    <li>python38Packages.shap</li>
    <li>python38Packages.skorch</li>
    <li>python38Packages.textacy</li>
    <li>python38Packages.toggl-cli</li>
    <li>python39Packages.baselines</li>
    <li>python39Packages.ignite</li>
    <li>python39Packages.labelbox</li>
    <li>python39Packages.ndjson</li>
    <li>python39Packages.optuna</li>
    <li>python39Packages.pyro-ppl</li>
    <li>python39Packages.pytorch-lightning</li>
    <li>python39Packages.pytorch-metric-learning</li>
    <li>python39Packages.pywick</li>
    <li>python39Packages.rising</li>
    <li>python39Packages.scancode-toolkit</li>
    <li>python39Packages.skorch</li>
    <li>python39Packages.textacy</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>28 packages failed to build:</summary>
  <ul>
    <li>apache-airflow</li>
    <li>backblaze-b2</li>
    <li>keepkey_agent (python38Packages.keepkey_agent)</li>
    <li>ledger_agent (python38Packages.ledger_agent)</li>
    <li>magic-wormhole (python38Packages.magic-wormhole)</li>
    <li>manim</li>
    <li>mnemosyne</li>
    <li>nrfutil</li>
    <li>python38Packages.datashader</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.mesa</li>
    <li>python38Packages.parfive</li>
    <li>python38Packages.pypiserver</li>
    <li>python38Packages.salmon-mail</li>
    <li>python38Packages.spacy</li>
    <li>python38Packages.sunpy</li>
    <li>trezor_agent (python38Packages.trezor_agent)</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.keepkey_agent</li>
    <li>python39Packages.ledger_agent</li>
    <li>python39Packages.magic-wormhole</li>
    <li>python39Packages.mesa</li>
    <li>python39Packages.parfive</li>
    <li>python39Packages.pypiserver</li>
    <li>python39Packages.salmon-mail</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.sunpy</li>
    <li>python39Packages.trezor_agent</li>
  </ul>
</details>
<details>
  <summary>133 packages built:</summary>
  <ul>
    <li>aiodnsbrute</li>
    <li>ansible-lint (python38Packages.ansible-lint)</li>
    <li>checkov</li>
    <li>conan</li>
    <li>dpt-rp1-py</li>
    <li>duplicity</li>
    <li>duply</li>
    <li>ffmpeg-normalize</li>
    <li>gdown (python38Packages.gdown)</li>
    <li>kaggle (python38Packages.kaggle)</li>
    <li>ldeep</li>
    <li>lieer</li>
    <li>luigi</li>
    <li>mathlibtools (python38Packages.mathlibtools)</li>
    <li>noto-fonts-emoji</li>
    <li>papis (python38Packages.papis)</li>
    <li>past-time</li>
    <li>piston-cli</li>
    <li>plex-mpv-shim</li>
    <li>python38Packages.afdko</li>
    <li>python38Packages.aiosmb</li>
    <li>python38Packages.ansible-kernel</li>
    <li>python38Packages.ansible-runner</li>
    <li>python38Packages.b2sdk</li>
    <li>python38Packages.cirq</li>
    <li>python38Packages.cve-bin-tool</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.dipy</li>
    <li>python38Packages.dnachisel</li>
    <li>python38Packages.enrich</li>
    <li>python38Packages.ffmpeg-progress-yield</li>
    <li>python38Packages.genome-collector</li>
    <li>python38Packages.gradient</li>
    <li>python38Packages.gradient-utils</li>
    <li>python38Packages.graphtage</li>
    <li>python38Packages.gtts</li>
    <li>python38Packages.habanero</li>
    <li>python38Packages.holoviews</li>
    <li>python38Packages.huggingface-hub</li>
    <li>python38Packages.hvplot</li>
    <li>python38Packages.hyperopt</li>
    <li>python38Packages.internetarchive</li>
    <li>python38Packages.libagent</li>
    <li>python38Packages.lsassy</li>
    <li>python38Packages.moviepy</li>
    <li>python38Packages.msldap</li>
    <li>python38Packages.myfitnesspal</li>
    <li>python38Packages.nagiosplugin</li>
    <li>python38Packages.nbsmoke</li>
    <li>python38Packages.nltk</li>
    <li>python38Packages.nototools</li>
    <li>python38Packages.panel</li>
    <li>python38Packages.papermill</li>
    <li>python38Packages.perfplot</li>
    <li>python38Packages.plexapi</li>
    <li>python38Packages.proglog</li>
    <li>python38Packages.pyppeteer</li>
    <li>python38Packages.pypykatz</li>
    <li>python38Packages.python-daemon</li>
    <li>python38Packages.python-miio</li>
    <li>python38Packages.python-mpv-jsonipc</li>
    <li>python38Packages.rbtools</li>
    <li>python38Packages.rich</li>
    <li>python38Packages.sacremoses</li>
    <li>python38Packages.surepy</li>
    <li>python38Packages.thinc</li>
    <li>python38Packages.tqdm</li>
    <li>python38Packages.transformers</li>
    <li>python38Packages.tubeup</li>
    <li>python38Packages.twine</li>
    <li>python39Packages.afdko</li>
    <li>python39Packages.aiosmb</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.ansible-lint</li>
    <li>python39Packages.ansible-runner</li>
    <li>python39Packages.b2sdk</li>
    <li>python39Packages.cirq</li>
    <li>python39Packages.cve-bin-tool</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.dipy</li>
    <li>python39Packages.dnachisel</li>
    <li>python39Packages.enrich</li>
    <li>python39Packages.ffmpeg-progress-yield</li>
    <li>python39Packages.gdown</li>
    <li>python39Packages.genome-collector</li>
    <li>python39Packages.gradient</li>
    <li>python39Packages.gradient-utils</li>
    <li>python39Packages.graphtage</li>
    <li>python39Packages.gtts</li>
    <li>python39Packages.habanero</li>
    <li>python39Packages.holoviews</li>
    <li>python39Packages.huggingface-hub</li>
    <li>python39Packages.hvplot</li>
    <li>python39Packages.hyperopt</li>
    <li>python39Packages.internetarchive</li>
    <li>python39Packages.kaggle</li>
    <li>python39Packages.libagent</li>
    <li>python39Packages.lsassy</li>
    <li>python39Packages.mathlibtools</li>
    <li>python39Packages.moviepy</li>
    <li>python39Packages.msldap</li>
    <li>python39Packages.myfitnesspal</li>
    <li>python39Packages.nagiosplugin</li>
    <li>python39Packages.nbsmoke</li>
    <li>python39Packages.nltk</li>
    <li>python39Packages.nototools</li>
    <li>python39Packages.panel</li>
    <li>python39Packages.papermill</li>
    <li>python39Packages.papis</li>
    <li>python39Packages.perfplot</li>
    <li>python39Packages.plexapi</li>
    <li>python39Packages.proglog</li>
    <li>python39Packages.pyppeteer</li>
    <li>python39Packages.pypykatz</li>
    <li>python39Packages.python-daemon</li>
    <li>python39Packages.python-miio</li>
    <li>python39Packages.python-mpv-jsonipc</li>
    <li>python39Packages.rbtools</li>
    <li>python39Packages.rich</li>
    <li>python39Packages.sacremoses</li>
    <li>python39Packages.surepy</li>
    <li>python39Packages.thinc</li>
    <li>python39Packages.tqdm</li>
    <li>python39Packages.transformers</li>
    <li>python39Packages.tubeup</li>
    <li>python39Packages.twine</li>
    <li>r128gain</li>
    <li>sacad</li>
    <li>spotdl</li>
    <li>theharvester</li>
    <li>tinyprog</li>
    <li>tockloader</li>
    <li>urlwatch</li>
  </ul>
</details>

**Pull request merged**